aviatrix/data_source_aviatrix_all_transit_gateways_test.go

```diff
@@ -0,0 +1,126 @@
+package aviatrix
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccDataSourceAviatrixAllTransitGateways_basic(t *testing.T) {
+	rName := acctest.RandString(5)
+	resourceName := "data.aviatrix_all_transit_gateways.foo"
+
+	skipAcc := os.Getenv("SKIP_DATA_ALL_TRANSIT_GATEWAY")
+	if skipAcc == "yes" {
+		t.Skip("Skipping Data Source All Transit Gateway tests as SKIP_DATA_ALL_TRANSIT_GATEWAY is set")
+	}
+	if skipAcc != "yes" {
+		gcpGwSize := os.Getenv("GCP_GW_SIZE")
+		if gcpGwSize == "" {
+			gcpGwSize = "n1-standard-1"
+		}
+		resource.Test(t, resource.TestCase{
+			PreCheck: func() {
+				testAccPreCheck(t)
+				preGatewayCheck(t, ". Set SKIP_DATA_ALL_TRANSIT_GATEWAY to yes to skip Data Source All Transit Gateway tests")
+			},
+			Providers: testAccProviders,
+			Steps: []resource.TestStep{
+				{
+					Config: testAccDataSourceAviatrixAllTransitGatewaysConfigPreBasic(rName),
+
+					Check: resource.ComposeTestCheckFunc(
+						resource.TestCheckResourceAttr("aviatrix_transit_gateway.test", "gw_name", fmt.Sprintf("aa-tfg-aws-%s", rName)),
+						resource.TestCheckResourceAttr("aviatrix_transit_gateway.test2", "gw_name", fmt.Sprintf("aa-tfg-gcp-%s", rName)),
+					),
+
+					Destroy: false,
+				},
+				{
+					Config: testAccDataSourceAviatrixAllTransitGatewaysConfigBasic(),
+					Check: resource.ComposeTestCheckFunc(
+						testAccDataSourceAviatrixAllTransitGateways(resourceName),
+						resource.TestCheckResourceAttr(resourceName, "transit_gateway_list.0.gw_name", fmt.Sprintf("aa-tfg-aws-%s", rName)),
+						resource.TestCheckResourceAttr(resourceName, "transit_gateway_list.0.vpc_id", os.Getenv("AWS_VPC_ID")),
+						resource.TestCheckResourceAttr(resourceName, "transit_gateway_list.0.vpc_reg", os.Getenv("AWS_REGION")),
+						resource.TestCheckResourceAttr(resourceName, "transit_gateway_list.0.gw_size", "t2.micro"),
+						resource.TestCheckResourceAttr(resourceName, "transit_gateway_list.1.gw_name", fmt.Sprintf("aa-tfg-gcp-%s", rName)),
+						resource.TestCheckResourceAttr(resourceName, "transit_gateway_list.1.gw_size", gcpGwSize),
+						resource.TestCheckResourceAttr(resourceName, "transit_gateway_list.1.account_name", fmt.Sprintf("aa-tfa-gcp-%s", rName)),
+						resource.TestCheckResourceAttr(resourceName, "transit_gateway_list.1.subnet", os.Getenv("GCP_SUBNET")),
+						resource.TestCheckResourceAttr(resourceName, "transit_gateway_list.1.vpc_reg", os.Getenv("GCP_ZONE")),
+					),
+					Destroy: true,
+				},
+			},
+		})
+	} else {
+		t.Log("Skipping Date Source All Transit Gateway test in AWS as SKIP_DATA_ALL_TRANSIT_GATEWAY_AWS is set")
+	}
+}
+
+func testAccDataSourceAviatrixAllTransitGatewaysConfigPreBasic(rName string) string {
+	gcpGwSize := os.Getenv("GCP_GW_SIZE")
+	if gcpGwSize == "" {
```

Comment:
If you are using a new environment variable for this test, in this case `GCP_GW_SIZE`, you should add it to test-infra/README_accep_test.md. You shouldn't need to hard code it here.

Reply:
add the request variable in test-infra/README_accep_test.md